### PR TITLE
Cpp polish 7

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: test/unit/module/real/core/is_gtz.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#include "test.hpp"
+#include <eve/constant/valmin.hpp>
+#include <eve/constant/valmax.hpp>
+#include <eve/function/is_gtz.hpp>
+#include <eve/constant/smallestposval.hpp>
+#include <eve/logical.hpp>
+#include <cmath>
```
Comment (jfalcou) [inline test/unit/module/real/core/is_gtz.cpp:14]: is `<cmath>` still needed ?

File: test/unit/module/real/core/is_infinite.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  TTS_EXPR_IS( eve::is_infinite(v_t())                  , logical<v_t> );
+};
+
+//==================================================================================================
+// Tests for eve::is_infinite
+//==================================================================================================
+
+EVE_TEST( "Check behavior of eve::is_infinite(simd)"
+        , eve::test::simd::ieee_reals
+        , eve::test::generate ( eve::test::ramp(0))
```
Comment (jfalcou) [inline test/unit/module/real/core/is_infinite.cpp:35]: this test looks like it doesn't test anything. We don't even test for eve::Inf/Minf.
We also test for all types YET we only test for IEEE reals there.

File: test/unit/module/real/core/is_less_equal.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+<typename T>(eve::as_<T>)
+{
+  using eve::logical;
+  using v_t = eve::element_type_t<T>;
+  TTS_EXPR_IS( eve::is_less_equal(T(), T())     , logical<T>   );
+  TTS_EXPR_IS( eve::is_less_equal(v_t(), v_t()) , logical<v_t> );
+  TTS_EXPR_IS( eve::is_less_equal(T(), v_t())   , logical<T> );
+  TTS_EXPR_IS( eve::is_less_equal(v_t(), T())   , logical<T> );
+
+  if constexpr(eve::floating_value<T>)
```
Comment (jfalcou) [inline test/unit/module/real/core/is_less_equal.cpp:33]: why this duplication ?

File: test/unit/module/real/core/is_nan.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+EVE_TEST( "Check behavior of eve::is_nan(simd)"
+        , eve::test::simd::ieee_reals
+        , eve::test::generate ( eve::test::ramp(0))
+        )
+<typename T>(T const& a0)
+{
+  using eve::detail::map;
+  using v_t = eve::element_type_t<T>;
+
+  TTS_EQUAL(eve::is_nan(a0), map([](auto e) -> eve::logical<v_t> { return  (e != e); }, a0));
```
Comment (jfalcou) [inline test/unit/module/real/core/is_nan.cpp:44]: can't we just test that's it's all false ?

**Review by jfalcou — CHANGES_REQUESTED** [review]
Some tests looks strange

File: test/unit/module/real/core/is_gtz.cpp (hunk 1 above)
Comment (jtlap) [reply test/unit/module/real/core/is_gtz.cpp:14]: suppressed

File: test/unit/module/real/core/is_infinite.cpp (hunk 2 above)
Comment (jtlap) [reply test/unit/module/real/core/is_infinite.cpp:35]: changed

File: test/unit/module/real/core/is_nan.cpp (hunk 4 above)
Comment (jtlap) [reply test/unit/module/real/core/is_nan.cpp:44]: almost does not work for non floating

**Review by jfalcou — APPROVED** [review]

**Pull request merged**